src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -25,27 +26,49 @@ class ReceiveComponent
             this.errorQueue = errorQueue;
         }
 
-        public void BindQueues(QueueBindings queueBindings)
+        public static ReceiveComponent Initialize(ReceiveConfiguration receiveConfiguration,
+            TransportInfrastructure transportInfrastructure,
+            PipelineComponent pipeline,
+            QueueBindings queueBindings,
+            EventAggregator eventAggregator,
+            CriticalError criticalError,
+            string errorQueue,
+            ReadOnlySettings settings)
         {
-            if (IsSendOnly)
-            {
-                return;
-            }
+            var receiveComponent = new ReceiveComponent(receiveConfiguration,
+                receiveConfiguration != null ? transportInfrastructure.ConfigureReceiveInfrastructure() : null, //don't create the receive infrastructure for send-only endpoints
+                pipeline,
+                eventAggregator,
+                criticalError,
+                errorQueue);
 
-            queueBindings.BindReceiving(configuration.LocalAddress);
+            receiveComponent.BindQueues(queueBindings);
 
-            if (configuration.InstanceSpecificQueue != null)
+            if (receiveConfiguration != null)
             {
-                queueBindings.BindReceiving(configuration.InstanceSpecificQueue);
+                settings.AddStartupDiagnosticsSection("Receiving", new
+                {
+                    receiveConfiguration.LocalAddress,
+                    receiveConfiguration.InstanceSpecificQueue,
+                    receiveConfiguration.LogicalAddress,
+                    receiveConfiguration.PurgeOnStartup,
+                    receiveConfiguration.QueueNameBase,
+                    TransactionMode = receiveConfiguration.TransactionMode.ToString("G"),
+                    receiveConfiguration.PushRuntimeSettings.MaxConcurrency,
+                    Satellites = receiveConfiguration.SatelliteDefinitions.Select(s => new
+                    {
+                        s.Name,
+                        s.ReceiveAddress,
+                        TransactionMode = s.RequiredTransportTransactionMode.ToString("G"),
+                        s.RuntimeSettings.MaxConcurrency
+                    }).ToArray()
+                });
             }
 
-            foreach (var satellitePipeline in configuration.SatelliteDefinitions)
-            {
-                queueBindings.BindReceiving(satellitePipeline.ReceiveAddress);
-            }
+            return receiveComponent;
         }
 
-        public async Task Initialize(ContainerComponent containerComponent, RecoverabilityComponent recoverabilityComponent)
+        public async Task PrepareToStart(ContainerComponent containerComponent, RecoverabilityComponent recoverabilityComponent)
```

Comment:
So this one is renamed to `PrepareToStart` because of the name collision with the recommended `Initialize` to be used to initialize any component. But then we have also a `Start` method for this component, which then collides with the component guidance to have Start/Stop methods. No?